### ISSUE TITLE
Error: 0x800703e3 when launching WSL [Insider, Build 17655, 17661 & 17666]

### ISSUE THREAD

**Issue by Seriell** [Microsoft/WSL#3134]
On the latest Skip Ahead Insider build (1803 / 10.0.17655.1000), WSL refuses to launch (On any distro), instead hanging (With Init using ~25% cpu at all times) before giving `Error: 0x800703e3` and sometimes `0x80070040`.
I've tried running `clean` and even installing a different distro (Can reproduce on Ubuntu and openSUSE) to no avail. 
I only noticed this happening directly after updating to 17655, and searching online for this error code only comes up with https://github.com/Microsoft/WSL/issues/1661, which doesn't appear to be related(?)

Full show of everything going on:
![image](https://user-images.githubusercontent.com/14004943/39407899-97c49af4-4bc5-11e8-8de2-5af9f49f088a.png)

**Comment by 0xADD1E** [comment]
I am encountering the same issue. Can anyone more familiar with the WSL internals give more information on further steps or determining steps to reproduce?

**Comment by 0xADD1E** [comment]
Additional information that may be useful: I'm seeing two init processes. One of them is using the entirety of a CPU core, the other of which is sitting idle. Killing the second one results in another printing of `Error: 0x80070490` whereas letting it time out results in the `Error: 0x800703e3`. In either case, a new idle init daemon is spawned.

Oddly enough, killing the init that's using the entire core will result in the idle one waking up, and taking up an entire core, and spawning it's own idle daemon. Killing the active one reverses it again and prints `Error: 0x80070040`

Hopefully this is helpful

**Comment by Biswa96** [comment]
Is that only for Windows Store Apps? Did anyone tried to install Linux userspace with Wsl* function?

**Comment by 0xADD1E** [comment]
I also tried it with https://github.com/yuk7/ArchWSL, to the same result, but I'm not familiar enough with lower-level WSL to know if that's what you mean.

With the behaviours of init, and the admittedly little that I know about WSL from Black Hat talks and blog posts, I'd guess the issue is with the re-implementation of init.

**Comment by Brian-Perkins** [comment]
Do either of you (@Seriell, @0xADD1E) happen to remember which build you were upgrading from?

**Comment by Seriell** [comment]
@Brian-Perkins Unfortunately I can't remember, it was either 17643 or 17650, either way it was from one Skip Ahead to another. I've been busy lately so I can't remember whether I skipped 17650 or not.
If theres some way in Windows to find out which I upgraded from, I would be more than happy to tell you.

**Comment by benhillis** [comment]
Could somebody experiencing these issues please collect and share logs so I can see what's going on?

https://github.com/Microsoft/WSL/blob/master/CONTRIBUTING.md#8-detailed-logs

**Comment by Brian-Perkins** [comment]
@Seriell I am sure there is an official way, but a good enough way is to find a "Build: " string in c:\windows.old\windows\logs\cbs\cbs.log (note the .old part of the path)

**Comment by Seriell** [comment]
@Brian-Perkins Unfortunately I was low on disk space earlier today and I let it delete Windows.old, sorry!

@benhillis Will do that now, hold up.

**Comment by Seriell** [comment]
@benhillis Here are the logs for the `Error: 0x800703e3` when launching WSL: [lxcore_logs_seriell.zip](https://github.com/Microsoft/WSL/files/1961483/lxcore_logs_seriell.zip)

All I did was run `ubuntu` like so:
![image](https://user-images.githubusercontent.com/14004943/39439496-b4a61d16-4c9f-11e8-8c0b-32bb649dbca1.png)

**Comment by 0xADD1E** [comment]
I had originally installed the latest non-insider build, and then upgraded to this build

**Comment by benhillis** [comment]
@Seriell - Thanks for sharing the logs. Unfortunately nothing in there that would indicate what is spinning.  Could you see if running bash.exe has the same behavior?

**Comment by Seriell** [comment]
bash.exe does the same thing, as does other distro exes. (I even tried as Admin but no dice) As noted by @0xADD1E it's probably some low level problem with the Init process.
I can provide more logs if needed, but I don't think the bash.exe ones would differ to just running `ubuntu`.

**Comment by Biswa96** [comment]
I tried to install a distro in fresh VM of Win build 17655. There is no problem with Wsl* functions (as was in 17650). Though I didn't try with Store app.
@Seriell Did you install the Ubuntu in previous Windows build? If there is no important files, can you uninstall Ubuntu from Store and try to reinstall it? Some problem shows in LX service. Try to restart the `LxssManager` service in service manager (services.msc).

**Comment by Seriell** [comment]
Yes I did install Ubuntu in a previous build (More specifically, 17643) from the Store app.
I don't imagine reinstalling Ubuntu from the store doing anything since I installed a fresh distro and that has the same problem, but I will try that as well as the services thing now. 
(I'll post a new comment when I've done that, didn't want to delay this information until I've tested stuff)

**Comment by benhillis** [comment]
@Seriell - Thanks. Unfortunately I think the next step would be to look at a full memory dump.  There are instructions for collecing one here: https://github.com/Microsoft/WSL/issues/2859#issuecomment-360350939

**Comment by Seriell** [comment]
@Biswa96 That service wasn't running when I first looked (Set to manual), I started it and nothing changed.

@benhillis I'll do that when I have time (Maybe later today, probably end up being tomorrow).
When I send the email should I reference this Issue in it?

**Comment by benhillis** [comment]
@Seriell - Thank you very much, yes please reference this issue and my Microsoft alias: benhill

**Comment by pryan1check** [comment]
i have same problem too. How to fix it?

**Comment by stoltzld** [comment]
I just reverted to 17650 and paused the insider updates.

**Comment by windrunner414** [comment]
i have same problem too.

**Comment by AnakinShieh** [comment]
I have tried the new build of Windows 17661, this problem still exists.

**Comment by lolmaus** [comment]
This error paralyzed my work. How do I recover from it?

**Comment by rodrymbo** [comment]
I discovered I was updated to 17661 when I logged in this morning. Getting this same error with either legacy or ubuntu. 

Anyone know if reverting to the previous build lets this work again?

**Comment by benhillis** [comment]
We still do not have enough information to fix this issue.  The most useful thing somebody that is experiencing this issue can do would be to take a kernel dump and share it with me via secure@microsoft.com.

**Comment by lolmaus** [comment]
@benhills, can you please provide a link to instructions?

PS I'm available on Gitter for realtime chat.

**Comment by benhillis** [comment]
@lolmaus - Instructions here https://github.com/Microsoft/WSL/issues/3134#issuecomment-385505538.

**Comment by rodrymbo** [comment]
Will any memory dump do, or just before/during/after trying to start wsl.exe, or after the error message shows up?

**Comment by benhillis** [comment]
@rodrymbo - It will have to be while the CPU is spinning trying to launch.

**Comment by rodrymbo** [comment]
Sending 110MB memory dump in process (slow upload).

**Comment by Seriell** [comment]
Apologies for the large delay but I've updated to 17661, verified the same thing is happening here and then sent you a memory dump via secure@microsoft.com 😃

**Comment by benhillis** [comment]
Thanks, I received one of your dumps (I'm not sure whos).  Question for everybody experiencing this issue, are you running an Intel or AMD processor?

**Comment by Seriell** [comment]
My dump would have been from my outlook email (Name should say "Erisa Arrowsmith" or otherwise have Erisa in it)
And uh, I do have an AMD processor, more specifically an AMD FX-4100

**Comment by stoltzld** [comment]
AMD A8

**Comment by benhillis** [comment]
I received two more dumps, thanks!  That should be enough to get this sorted out.

**Comment by windrunner414** [comment]
AMD Ryzen 1700 and Ryzen 2700X

**Comment by lolmaus** [comment]
AMD Ryzen 2700X.

Wasn't able to take a memory dump. 😔 Both my keyboards lack the ScrollLock button. Tried remapping and failed miserably.

**Comment by rodrymbo** [comment]
Mine's also an AMD.

**Comment by 0xADD1E** [comment]
Issue not fixed on 17661

AMD Ryzen 2600

**Comment by benhillis** [comment]
We have a local repro, thanks to everybody who has sent us dumps. We are working on getting a fix out ASAP.

**Comment by Biswa96** [comment]
Did you find out the cause? Can you share it?

**Comment by Seriell** [comment]
After being unable to use WSL for a week because of this issue, I too am curious as to what caused it 😛

**Comment by rodrymbo** [comment]
I keep clicking on shortcuts I've made to WSL bash scripts, forgetting that nothing will happen. I can do all those things manually of course, but it's a lot less work with the scripts. So, anyway, I can suppress my curiosity about how it happened as long as it gets fixed!

I suppose I could use this as an excuse to figure out how to do some or all of those scripts in powershell. 😛

**Comment by benhillis** [comment]
Issue is still being debugged, when I have more info I will share.

**Comment by littleq0903** [comment]
same here, AMD Ryzen 1700

**Comment by DaJakerBoss** [comment]
AMD A8-4500M APU

Was working fine on 17655, but then my HDD went out. Replaced the HDD with an SSD yesterday and upgraded my RAM from 6GB 1600MHz to 8GB 1066MHz, installed Windows 10 Pro fresh, then updated to 17661 last night. 

Today I've reinstalled Ubuntu multiple times and nothing is working. I am noting that several of my applications are reporting memory violations every single time I open them. Spotify Installer, Avast, F.lux, and others.

Will try a RAM downgrade to see if it is linked to AMD CPU's communicating with RAM.

Quick note, I did have to unblock `init` after it had been previously blocked by Comodo Antivirus' HIPS tool. Disabled and re-installed Ubuntu to no avail.

Hoping y'all can iron this one out quick 😄

**Comment by DaJakerBoss** [comment]
Thank you @benhillis !!!!!

**Comment by Biswa96** [comment]
What about the Spring CU? Normal people don't use insider builds. And how this issue only affects AMD CPU? Which one-- LxCore or LxssManager?

**Comment by DaJakerBoss** [comment]
Actually @benhillis could this issue go beyond WSL? I'm getting multiple Memory Violation (0xc0000005) errors on several different applications. Bad enough to force me to roll back.

**Comment by benhillis** [comment]
Yes, the issue is in the Windows memory manager and not specific to WSL.

**Comment by benhillis** [comment]
@Biswa96 - this bug is not present in 1803.

**Comment by Seriell** [comment]
I haven't found any memory issues in anything other than WSL, but maybe I didnt look hard enough.
And yeah, this is only in insider builds.

**Comment by DaJakerBoss** [comment]
I've documented my findings [here](https://answers.microsoft.com/en-us/insider/forum/insider_wintp-insider_perf-insiderplat_pc/memory-violations-0xc0000005-upon-opening-several/a6be9c63-492e-403a-a84e-328fdb153150) 

No more errors after I've rolled back.

**Comment by DaJakerBoss** [comment]
I fell back to stable windows but (thanks Windows) was automatically
updated to the insider build again. Apps did not improve.

On Mon, May 7, 2018 at 4:24 PM, IOWEB TECHNOLOGIES <notifications@github.com

**Comment by benhillis** [comment]
@DaJakerBoss  - you'll have to select "Stop Insider Preview builds" via Settings -> Update and Security -> Windows Insider Program.

**Comment by DaJakerBoss** [comment]
Yes, I understand. Error on my behalf :-P

Thank You!
Jake Hauser

*Short messages are not meant to display lack of interest for me, but
should rather be interpreted as knowledge of the subject at hand.*

*"If you can't explain it simply,*
*you don't understand it well enough."*
*--Albert Einstein*

**Comment by stoltzld** [comment]
You can use the pause feature too.  I just wish it lasted longer than 7 days.

**Comment by DaJakerBoss** [comment]
I am using the pause feature. nice feature actually :)

Why do you wish it lasted longer?

**Comment by stoltzld** [comment]
So I can 16650 having to fiddle.

**Comment by DaJakerBoss** [comment]
> So I can 16650 having to fiddle.

What exactly does that mean?

**Comment by stoltzld** [comment]
I guess it means that I'm too lazy to finish typing what I meant to say before I hit the comment button *laugh*

**Comment by DaJakerBoss** [comment]
I don't blame you. It would have happened to me too

Thank You!
Jake Hauser

*Short messages are not meant to display lack of interest for me, but
should rather be interpreted as knowledge of the subject at hand.*

*"If you can't explain it simply,*
*you don't understand it well enough."*
*--Albert Einstein*

**Comment by CajuCLC** [comment]
Late to the party, but I went from 10.0.17134.1 to 10.0.17661. I haven't seen any problem with Windows itself, only WSL (Ubuntu).
AMD Ryzen 1800X here.

**Comment by DaJakerBoss** [comment]
I never had any problems when I did the same, but then I had to reinstall windows and that's where the issues came in. It probably change from device to device

**Comment by Seriell** [comment]
New build released (17666) - @benhillis is it safe to update to this or is the issue still present?

**Comment by benhillis** [comment]
17666 does not have the fix.

**Comment by DaJakerBoss** [comment]
Are there any fixes to windows as a whole? Any fixes for the better if applications?

**Comment by Seriell** [comment]
(I updated the title so people don't come in asking if its fixed in 17666, although I doubt it makes a difference heh)

**Comment by moisesmcardona** [comment]
On 17666, I get `The I/O operation has been aborted because of either a thread exit or an application request.`. Using AMD Ryzen 2700X

**Comment by DaJakerBoss** [comment]
@Moises Cardona is this in WSL?

Thank You!
Jake Hauser

*Short messages are not meant to display lack of interest for me, but
should rather be interpreted as knowledge of the subject at hand.*

*"If you can't explain it simply,*
*you don't understand it well enough."*
*--Albert Einstein*

**Comment by moisesmcardona** [comment]
![image](https://user-images.githubusercontent.com/16326634/39893890-384fa7c4-5473-11e8-897a-8a522b931bfe.png)

Yes @DaJakerBoss

**Comment by benhillis** [comment]
@moisesmcardona - That is the same error. In the most recent Insider build, wsl.exe and bash.exe will print error strings instead of hex error values whenever possible.

**Comment by DaJakerBoss** [comment]
@Ben Hillis will the bug fix for WSL also fix some of the other
memory-related issues? 17666 is still having the same application errors as
before (though I'm not surprised since there was no fix for WSL in 666
either).

Speaking of, why isn't this issue listed in the Insider Blog?

**Comment by benhillis** [comment]
Yes it should. I'm not sure, I'll sync up with the people who write those release notes tomorrow and see about adding something.

**Comment by DaJakerBoss** [comment]
thank you so much!

**Comment by ioweb-gr** [comment]
Is it fixed in 17672.1000 perhaps?

**Comment by benhillis** [comment]
@ioweb-gr - Yep! This is fixed in 17672.

**Comment by ioweb-gr** [comment]
That's great to hear :)

**Comment by SemperFu** [comment]
Just tried it and it works!

**Comment by DaJakerBoss** [comment]
Thank you so much! I can get back to work now!

**Comment by ioweb-gr** [comment]
I just updated and it works fine for me too.

**Comment by hmqgg** [comment]
17672 Fixed. Thanks!

**Comment by Seriell** [comment]
Updated and can confirm it works here too.
Since this originated in Insider builds and was fixed in insider too, may I close this issue?

**Comment by stoltzld** [comment]
Thanks.

**Comment by benhillis** [comment]
Thanks for confirming the issue is resolved.

**Comment by arcsoftech** [comment]
The error started poping up again in April 2020 update

**Comment by Erisa** [comment]
> 
> 
> The error started poping up again in April 2020 update

I don't have this issue on build 19041.207 with Ryzen 7 2700X, it may be something else causing it. Might also help to know what your CPU is since the original isssue only affected AMD CPUs.
Remember that the error code just means `The I/O operation has been aborted because of either a thread exit or an application request..`, which can be caused from a number of things.

**Comment by gmos** [comment]
@arcsoftech  Same for me.  It happens during startup, before anything else is displayed.  And I am on an Intel CPU.

**Comment by gmos** [comment]
@Erisa Perhaps re-open the issue or should we make a new?

**Comment by Erisa** [comment]
> @Erisa Perhaps re-open the issue or should we make a new?

If it's affecting Intel CPUs then it's probably not relayed to this problem and you should probably open a new issue.

**Comment by MBetters** [comment]
I got this issue as well. I found that the most stable way to launch WSL Bash is to wait until after these two conditions are met:

- My laptop has connected to all my network drives (company VPN done init, etc)
- My Docker Desktop Hyper-V has finished initializing. (The `Docker desktop is running. Open PowerShell and start hacking with docker or compose` notification has popped up)

Once those two conditions are met, I open my PowerShell 7 and type `bash` and don't get the error noted above.

My Winver:
![image](https://user-images.githubusercontent.com/12706574/83036941-2ee30080-a009-11ea-83e4-05c4dc524630.png)
